libs/core/src/components/badge/badge.component.scss

```diff
@@ -35,6 +35,14 @@
   }
 }
 
+:host-context(.segment-btn-wrapper) {
+  --kirby-badge-elevation: #{get-elevation(2)};
+  position: absolute;
+  top: -#{size('xxs')};
+  right: #{size('xxs')};
+  z-index: z('segmentBadge');
+}
```

Comment:
I'm not a fan of "decoupled context dependant styles" (as you probably know 😊)

Reply:
btw., do you know, what the thing is with `:host-context()` outside of Angular? It seems to have poor support (Chromium-only) https://caniuse.com/?search=%3Ahost-context()